sys/kern/kern_cheri_revoke.c

```diff
@@ -146,6 +146,57 @@ static SYSCTL_NODE(_kern_proc, KERN_PROC_QUARANTINING, quarantining,
     CTLFLAG_RD | CTLFLAG_MPSAFE, sysctl_kern_proc_quarantining,
     "is this process quarantining for temporal safety");
 
+static int
+sysctl_kern_proc_revoker_state(SYSCTL_HANDLER_ARGS)
+{
+	struct vmspace *vm;
+	int *name = (int *)arg1;
+	u_int namelen = arg2;
+	struct proc *p;
+	int error;
+	pid_t pid;
+	int state;
+
+	if (namelen != 1)
+		return (EINVAL);
+
+	pid = (pid_t)name[0];
+	p = curproc;
+	if (pid == p->p_pid || pid == 0) {
+		if (!SV_CURPROC_FLAG(SV_CHERI)) {
+			state = -1;
+			goto out_nocheri;
+		}
+
+		vm = vmspace_acquire_ref(p);
```

Comment:
Same comments as in the other diff regarding acquiring a ref for one's own vmspace.